webpack/assets/javascripts/react_app/components/TemplateGenerator/__tests__/TemplateGeneratorActions.test.js

```diff
@@ -0,0 +1,83 @@
+import { saveAs } from 'file-saver';
+import API from '../../../API';
+import { runActionInDepth } from '../../../common/testHelpers';
+
+import {
+  dataUrl,
+  failResponseMock,
+  scheduleResponse,
+  noContentResponse,
+  generatedReportResponse,
+} from './TemplateGenerator.fixtures';
+import {
+  TEMPLATE_GENERATE_REQUEST,
+  TEMPLATE_GENERATE_POLLING,
+  TEMPLATE_GENERATE_SUCCESS,
+  TEMPLATE_GENERATE_FAILURE,
+} from '../TemplateGeneratorConstants';
+import * as actions from '../TemplateGeneratorActions';
+
+jest.mock('file-saver');
+jest.mock('../../../API');
+
+beforeEach(() => {
+  API.post.mockImplementation(async () => scheduleResponse);
+  API.get.mockImplementation(async () => noContentResponse);
+});
+
+describe('TemplateGeneratorActions', () => {
+  beforeEach(() => {
+    API.post.mockClear();
+    API.get.mockClear();
+  });
+
+  describe('generateTemplate', () => {
+    it('schedule generation', () => {
+      runActionInDepth(() =>
+        actions.generateTemplate('/schedule', { foo: 'bar' })
+      ).then(callTree => {
+        expect(callTree[0].type).toEqual(TEMPLATE_GENERATE_REQUEST);
+        expect(callTree[0].payload).toHaveProperty('foo', 'bar');
+        expect(API.post).toHaveBeenCalledWith('/schedule', { foo: 'bar' });
+      });
+    });
+
+    it('starts polling', () => {
+      runActionInDepth(() => actions.generateTemplate(), 2).then(callTree => {
+        expect(callTree[1][0].type).toEqual(TEMPLATE_GENERATE_POLLING);
+        expect(callTree[1][0].payload).toHaveProperty('url', dataUrl);
+        expect(API.get.mock.calls[0]).toEqual([dataUrl, expect.any(Object)]);
+      });
+    });
+
+    it('repeats polling till result ready', () => {
+      API.get
+        .mockImplementationOnce(async () => noContentResponse)
+        .mockImplementationOnce(async () => generatedReportResponse);
+
+      runActionInDepth(() => actions.generateTemplate(), 3).then(callTree => {
```

Comment:
Here is example usage of the new test helper.

Reply:
wouldn't `testActionSnapshotWithFixtures()` test the same with a snapshot?

Reply:
It would not, because it is generating calls only in depth of 1, so if dispatch is called with another dispatch function, and from that again, the snapshot won't have that covered, just will state function in the tree. Another thing is, that this is more explicit, than just a snapshot, I prefer to be more explicit in the tests (otherwise we can update the testActionSnapshotFixtures to take the depth argument, but not in this PR i guess)